RoslynPluginGenerator/AnalyzerPluginGenerator.cs

```diff
@@ -67,6 +73,10 @@ public bool Generate(NuGetReference analyzeRef, string language, string sqaleFil
             }
             SupportedLanguages.ThrowIfNotSupported(language);
 
+            RoslynPluginDefinition definition = new RoslynPluginDefinition();
```

Comment:
The "BuildPlugin" method had a lot of parameters and this change would have meant adding more. I've introduced this new data class to hold the data required to describe a Roslyn plugin.
I'll move these lines down to join up with those starting in line 96 so all of the assignments to values in the class are in one place.